crates/motoko/src/lib/vm.rs

```diff
@@ -212,15 +216,21 @@ fn call_prim_function(
                 Ok(Step {})
             }
             v => {
-                let mut txt = Vector::new();
-                txt.push_front(string_from_value(&v)?);
-                let txt = crate::value::Text(txt);
+                let txt = string_from_value(&v)?;
                 log::info!("DebugPrint: {}: {:?}", core.cont_source, txt);
-                core.debug_print_out.push_back(txt);
+                core.debug_print_out.push_back(txt.into());
                 core.cont = Cont::Value(Value::Unit);
                 Ok(Step {})
             }
         },
+        OpenValue => {
+            core.cont = Cont::Value(args.to_motoko().map_err(Interruption::ValueError)?);
+            Ok(Step {})
+        }
+        CloseValue => {
+            core.cont = Cont::Value(args.to_rust::<Value>().map_err(Interruption::ValueError)?);
+            Ok(Step {})
+        }
```

Comment:
These primitives are something I've wanted in Motoko for a long time, but with the type system, it wasn't clear how best to type them (though I can think of a few options).

Now, I'm wondering what best to call them?

Rather than "open" and "close", maybe "into" and "from"?  (`IntoValue` and `FromValue`).  Though, "open" and "close" also makes sense as well.